tests/js/session-tests.js

```diff
@@ -527,6 +527,24 @@ module.exports = {
       .then(() => TestCase.assertTrue(progressCalled));
   },
 
+  // test that it is possible to register a custom logging function on a
+  // Sync session, and that it is invoked when a session is established
+  async testCustomSessionLogging() {
+    const partition = Utils.genPartition();
+    let credentials = Realm.Credentials.anonymous();
+    let app = new Realm.App(appConfig);
+    let user = await app.logIn(credentials);
+    let config = getSyncConfiguration(user, partition);
+
+    let loggerInvoked = false;
+    Realm.App.Sync.setLogLevel(app, "info"); // "all", "trace", "debug", "detail", "info", "warn", "error", "fatal", "off"
+    Realm.App.Sync.setLogger(app, (level, message) => (loggerInvoked = true));
+    const realm = await Realm.open(config);
+    realm.close();
+
+    TestCase.assertTrue(loggerInvoked, "Custom logger should have been invoked");
```

Comment:
I agree, this could instead await the logger callback being called.

```suggestion
    const promisedLog = new Promise(resolve => {
        Realm.App.Sync.setLogLevel(app, "info"); // "all", "trace", "debug", "detail", "info", "warn", "error", "fatal", "off"
        Realm.App.Sync.setLogger(app, (level, message) => resolve(message));
    });
    const realm = await Realm.open(config);
    const message = await promisedLog;
    realm.close();

    TestCase.assertEqual(message, /* whatever the first message will be */);
```

Reply:
I'm not sure we can say for certain what the first message will be.  Maybe we can just `resolve(true)`?

Reply:
If we don't know for sure what the first message is then I think the current way seems fine 👍